app/services/invoices/payments/retry_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Invoices
+  module Payments
+    class RetryService < BaseService
+      def initialize(invoice:)
+        @invoice = invoice
+
+        super
+      end
+
+      def call
+        return result.not_found_failure!(resource: 'invoice') if invoice.blank?
+        return result.not_allowed_failure!(code: 'invalid_status') if invoice.draft? || invoice.succeeded?
+
+        unless invoice.ready_for_payment_processing?
+          return result.not_allowed_failure!(code: 'payment_processor_is_currently_handling_payment')
+        end
+
+        webhook_type = invoice.invoice_type == 'subscription' ? (:invoice) : (invoice.invoice_type.to_sym)
```

Comment:
Style/TernaryParentheses: Use parentheses for ternary expressions with complex conditions.